### DISCUSSION TITLE
Start menu shortcut

### DISCUSSION THREAD

**Discussion by telmobss1** [lxqt/lxqt#1947] [Q&A]
![IMG_20210312_123538](https://user-images.githubusercontent.com/80489312/110941128-bc40d880-832f-11eb-8615-8e0c4bd083ce.jpg)

Shortcuts are important, especially if some shortcuts are already configured by default

What needs to be done for the Windows super key to become a shortcut to my applications?
That in addition to being practical, many Windows people are already used to it, and that way it would facilitate the migration for those people who have a less powerful PC

**Comment by stefonarch** [comment]
Edit: ignore this, not necessary.

There is a trick to do so, but when WinKey is bound to the menu it cannot be used anymore for as modifier.
 Create a hidden file `~/.winkey`
 
Insert

```
#!/bin/bash
export DISPLAY=:0.0 ;xmodmap -e 'keycode 133 = F13'
```
make it executable and add it to startup applications in session settings, relogin, then press  winkey as shortcut in the menu configuration.

**Reply by telmobss1** [reply]
it would be good at least, through the graphical interface in the settings of the shortcuts to have predefined shortcuts


Anyway it is an idea that I would like to have come by default in the lxqt interface to make it easier for users who do not understand anything about Linux

**Reply by tsujan** [reply]
I may have not understood what the problem is but there is no binding relationship between the `Meta` key and Start Menu. You could give another shortcut to Start Menu in LXQt Shortcut Keys (aka. Global Action Manager); anyone not familiar with Linux can do that.

**Reply by telmobss1** [reply]
that's exactly what i wanted to happen

Like KDE, gnome and Cinnamom, it would be a good idea to set the super meta key as the default key to open the application menu by default

**Reply by tsujan** [reply]
@telmo1241 
I don't think there's a problem to solve. You can already assign the `Meta` (aka. `Super` or `Windows`) key to the Start Menu — although it isn't a good choice because `Meta` can be used in customized shortcuts of various apps.

**Reply by stefonarch** [reply]
As I said, there could be issues with other shortcuts, for example for keyboard layout switching there are many using the win key.
![schermata-13-59-40](https://user-images.githubusercontent.com/10681413/110943184-34f96200-833b-11eb-9379-8866cf306b13.png)

They are in the code and it's not possible to add a custom one for now.

**Reply by tsujan** [reply]
> for example for keyboard layout switching there are many using the win key.

So, if they want to use it elsewhere, they'll need to change it there (I use the "Menu" key). That's the case everywhere: you can't have ambiguous shortcuts.

**Reply by stefonarch** [reply]
> You can already assign the `Meta` (aka. `Super` or `Windows`) key to the Start Menu

No, it's not possible to add it directly atm, similar to as it's not possible to add Alt or Ctrl.

**Reply by tsujan** [reply]
I already did. It was registered as `Super_L` and worked fine.

EDIT: I also found a bug in a recent commit; will fix it in a few minutes.

**Reply by stefonarch** [reply]
So I made some confusion here, sorry.
It's no problem adding the WinKey to the menu using the settings, and in fact for example  Lubuntu does this by default.

I'll move this to Q&A.

**Reply by tsujan** [reply]
> and in fact for example Lubuntu does this by default.

Which is a terrible choice because it disables `Meta` in apps (as KDE does).

**Reply by stefonarch** [reply]
I'm testing winkey for menu on debian now, and I noticed that I can well use Meta+1|2|3 to go through the panel items.
It looks like the menu reacts only to a release of `Super_L` - just pressing it doesn't open the menu. So it should still work for all other shortcuts.

**Reply by tsujan** [reply]
No (custom) shortcut with `Meta` works in FeatherPad with it. Either the user shouldn't set `Meta` for the main menu or he should forget about it in custom shortcuts of apps.

We should fix the problem at some point.

**Reply by tsujan** [reply]
@stefonarch 
Very unrelated to this: Your membership in LXQt isn't shown in your comments when one isn't signed into GitHub. I think you may want to enable LXQt in your GitHub setting -- I don't remember where. Of course, that's optional.

**Reply by stefonarch** [reply]
That is only here  in discussions like this  (membership) AFAIK, I'm always logged in.
And yes, meta shortcuts are only working for the panel application activating and keyboard layout switching and nowhere else.

**Reply by tsujan** [reply]
> That is only here in discussions like this

I saw it at https://github.com/lxqt/lxqt-globalkeys/pull/210, without signing into GitHub (it says "Contributor"). It isn't important; just thought you might want to know.

> And yes, meta shortcuts are only working for...

I think a fix is possible in the current code; should find time to test.

**Reply by stefonarch** [reply]
Oh, now I see, using a private firefox window. 
Went trough my settings but didn't find anything which changes that "collaborator", also "Organisations" section  is not shown on my profile for non logged users, but I don't mind.

**Reply by tsujan** [reply]
I was curious about it and remembered that my membership was "private"; changed it to "public". This is outdated but may work: https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership

**Reply by wxl** [reply]
> > and in fact for example Lubuntu does this by default.

> Which is a terrible choice because it disables `Meta` in apps (as KDE does).

It certainly has its drawbacks. Since you mention KDE, apparently there's a [trick](https://userbase.kde.org/Plasma/Tips#Windows.2FMeta_Key) to dealing with <kbd>Meta</kbd> in Plasma. Just FYI.

**Reply by stefonarch** [reply]
> This is outdated but may work

In fact it does, but I never would have found it :)
